apps/client/src/routes/dashboards/dashboards-index/components/delete-dashboard-modal.tsx

```diff
@@ -81,132 +83,138 @@ export function DeleteDashboardModal(props: DeleteDashboardModalProps) {
-            values={{
-              b: (n) => (
-                <Box variant="span" fontWeight="bold">
-                  {n}
-                </Box>
-              ),
-              name: props.dashboards.at(0)?.name ?? 'Loading...',
-            }}
-          >
-            {(message) => <Box variant="span">{message}</Box>}
-          </FormattedMessage>
-        ) : (
-          <FormattedMessage
-            defaultMessage="Permanently delete <b>{count} dashboards</b>? You cannot undo this action."
-            description="delete dashboard modal delete dashboards message"
-            values={{
-              b: (c) => (
-                <Box variant="span" fontWeight="bold">
-                  {c}
-                </Box>
-              ),
-              count: props.dashboards.length,
-            }}
-          >
-            {(message) => <Box variant="span">{message}</Box>}
-          </FormattedMessage>
-        )}
+            </Link>
+          </Alert>
 
-        <Alert type="warning" statusIconAriaLabel="Warning">
-          <FormattedMessage
-            defaultMessage="Proceeding with this action will delete the dashboard with all its content and can affect related resources."
-            description="delete dashboard modal warning message"
-          />
-          <Link
-            external={true}
-            href="https://github.com/awslabs/iot-application"
-          >
+          <Box>
             <FormattedMessage
-              defaultMessage="Learn more"
-              description="delete dashboard modal learn more link"
+              defaultMessage="To avoid accidental deletions, we ask you to provide additional written consent."
+              description="delete dashboard modal consent request"
             />
-          </Link>
-        </Alert>
+          </Box>
 
-        <Box>
-          <FormattedMessage
-            defaultMessage="To avoid accidental deletions, we ask you to provide additional written consent."
-            description="delete dashboard modal consent request"
-          />
-        </Box>
+          <ColumnLayout columns={2}>
+            <form
+              onSubmit={(event) => {
+                event.preventDefault();
 
-        <ColumnLayout columns={2}>
-          <form
-            onSubmit={(event) => {
-              event.preventDefault();
+                void handleSubmit(() => {
+                  handleDelete();
+                })();
+              }}
+            >
+              <Form>
+                <Controller
+                  name="consent"
+                  control={control}
+                  rules={{ required: true, pattern: /^confirm$/ }}
+                  render={({ field }) => (
+                    <FormField>
+                      <Input
+                        ariaRequired
+                        placeholder={DELETE_CONSENT_TEXT}
+                        onChange={(event) => field.onChange(event.detail.value)}
```

Comment:
This would probably benefit from having a `debounce` implemented. Depending on requirements, you could consider using `onBlur` instead of `onChange` to get the same effect with built-in functionality.

Reply:
The component follows the Cloudscape UX pattern, which relies on an onChange event. See the behavior of https://cloudscape.design/examples/react/delete-with-additional-confirmation.html in the demo for the pattern.

I agree with your rationale, but it would not enable the expected UX to use a blur event. Likewise, a debounce would impact the immediacy of enabling the "Confirm" button. All of this logic is handled by [`react-hook-form`](https://react-hook-form.com/), which is configured to manage re-renders efficiently.